utils/guard.go

```diff
@@ -0,0 +1,39 @@
+package utils
```

Comment:
Thanks for adding this

Reply:
Glad you see value in this. It's not as easy as C++/RAII unfortunately. Even after I wrote this I forgot I had to explicitly write `defer guard.OnFail()`. So that was fun.